src/components/application_manager/src/message_helper.cc

```diff
@@ -2394,41 +2394,39 @@ mobile_apis::Result::eType MessageHelper::ProcessSoftButtons(
       return Result::DISALLOWED;
     }
 
+    bool is_image_valid = false;
+    bool is_image_exists = request_soft_buttons[i].keyExists(strings::image);
+
+    bool is_text_valid = false;
+    bool is_text_exists = request_soft_buttons[i].keyExists(strings::text);
+
+    if (is_image_exists) {
+      is_image_valid =
```

Comment:
@indlu I think it will be even better to rename `request_soft_buttons` -> `soft_buttons` and then inside loop create separate variables: `soft_buttons_text = soft_buttons[strings::text]`, `soft_buttons_img = soft_buttons[strings::img]` and so on.